url11fb0973cf_raichuman9_Swimmer526_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/org/jbox2d/dynamics/World.java

```diff
@@ -0,0 +1,1400 @@
+					
+					// Don't simulate joints connected to inactive bodies.
+					if (other.isActive() == false) {
+						continue;
+					}
+					
+					island.add(je.joint);
+					je.joint.m_islandFlag = true;
+					
+					if ((other.m_flags & Body.e_islandFlag) == Body.e_islandFlag) {
+						continue;
+					}
+					
+					assert (stackCount < stackSize);
+					stack[stackCount++] = other;
+					other.m_flags |= Body.e_islandFlag;
+				}
+			}
+			
+			island.solve(step, m_gravity, m_allowSleep);
+			
+			// Post solve cleanup.
+			for (int i = 0; i < island.m_bodyCount; ++i) {
+				// Allow static bodies to participate in other islands.
+				Body b = island.m_bodies[i];
+				if (b.getType() == BodyType.STATIC) {
+					b.m_flags &= ~Body.e_islandFlag;
+				}
+			}
+		}
+		
+		// Synchronize fixtures, check for out of range bodies.
+		for (Body b = m_bodyList; b != null; b = b.getNext()) {
+			// If a body was not in an island then it did not move.
+			if ((b.m_flags & Body.e_islandFlag) == 0) {
+				continue;
+			}
+			
+			if (b.getType() == BodyType.STATIC) {
+				continue;
+			}
+			
+			// Update fixtures (for broad-phase).
+			b.synchronizeFixtures();
+		}
+		
+		// Look for new contacts.
+		m_contactManager.findNewContacts();
+	}
+	
+	private void solveTOI() {
+		// Prepare all contacts.
+		for (Contact c = m_contactManager.m_contactList; c != null; c = c.m_next) {
+			// Enable the contact
+			c.m_flags |= Contact.ENABLED_FLAG;
+			
+			// Set the number of TOI events for this contact to zero.
+			c.m_toiCount = 0;
+		}
+		
+		// Initialize the TOI flag.
+		for (Body body = m_bodyList; body != null; body = body.m_next) {
+			// Kinematic, and static bodies will not be affected by the TOI event.
+			// If a body was not in an island then it did not move.
+			if ((body.m_flags & Body.e_islandFlag) == 0 || body.getType() == BodyType.KINEMATIC
+					|| body.getType() == BodyType.STATIC) {
+				body.m_flags |= Body.e_toiFlag;
+			}
+			else {
+				body.m_flags &= ~Body.e_toiFlag;
+			}
+		}
+		
+		// Collide non-bullets.
+		for (Body body = m_bodyList; body != null; body = body.m_next) {
+			if ((body.m_flags & Body.e_toiFlag) == Body.e_toiFlag) {
+				continue;
+			}
+			
+			if (body.isBullet() == true) {
+				continue;
+			}
+			
+			solveTOI(body);
+			
+			body.m_flags |= Body.e_toiFlag;
+		}
+		
+		// Collide bullets.
+		for (Body body = m_bodyList; body != null; body = body.m_next) {
+			if ((body.m_flags & Body.e_toiFlag) == Body.e_toiFlag) {
+				continue;
+			}
+			
+			if (body.isBullet() == false) {
+				continue;
+			}
+			
+			solveTOI(body);
+			
+			body.m_flags |= Body.e_toiFlag;
+		}
+	}
+	
+	private final TOIInput toiInput = new TOIInput();
+	private final TOIOutput toiOutput = new TOIOutput();
+	private final Sweep backup = new Sweep();
+	private final TOISolver toiSolver = new TOISolver();
+	
+
+	private Contact[] m_contacts = new Contact[Settings.maxTOIContacts];
+	
+	private void solveTOI(Body body) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 33. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 1184-1191, lines 1196-1260 into separate methods.